src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedString = '';
+    let elementSeparator = separator + '';
+
+    if (separator === undefined) {
+      elementSeparator = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === null || element === undefined) {
+        element = '';
+      }
+
+      if (i === this.length - 1) {
+        joinedString += element;
+      } else {
+        joinedString += element + elementSeparator;
```

Comment:
your approach is working, but if you use @Dima-Batiuk suggestion, you can avoid extra variable `element` and extra check just use 
```
if (this[i] !== null && this[i] !== undefined) {
  // add element to result
}
```
and after that could be a second check, if that's not the last element add separator